### PR TITLE
Add Endpoints 1.1 sample

### PR DESCRIPTION
**Pull request by saicheems**
Updated the sample to use the Endpoints 1.1 Python client with
configuration options for service management and service control.

### CODE REVIEW

**Comment by saicheems** [comment]
This is an in progress sample for the 1.1 client - The functions include /echo and /echo/getUserEmail. We're also trying to show off integration with service control and service management via swagger.

Any feedback on the design and how to improve the surface is welcome.

File: appengine/standard/endpoints/backend/app.yaml [hunk 1]
```diff
@@ -3,14 +3,25 @@ threadsafe: true
...
 
 libraries:
 - name: pycrypto
   version: 2.6
-- name: endpoints
-  version: 1.0
+
+# Endpoints 1.1 uses background threads for caching and reporting to service
+# management and service control.
+manual_scaling:
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/app.yaml]: Ouch. Is there any plans to make this work with auto-scaled modules?

File: appengine/standard/endpoints/backend/app.yaml [hunk 2]
```diff
@@ -3,14 +3,25 @@ threadsafe: true
...
+  instances: 1
+
+beta_settings:
+  use_endpoints_api_management: true
+  endpoints_swagger_spec_file: echo-v1_swagger.json
+
+env_variables:
+  #TODO: The Endpoints service name.
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/app.yaml]: Don't use #TODO, just say "Replace with your endpoints service name, for example your-service.appspot.com"

File: appengine/standard/endpoints/backend/main.py [hunk 3]
```diff
@@ -22,128 +22,46 @@
 from protorpc import remote
 # [END imports]
 
-
 # [START messages]
-class Greeting(messages.Message):
-    """Greeting that stores a message."""
+class Echo(messages.Message):
+    """An echo that stores a message."""
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/main.py]: How about `A proto Message that contains a simple string field.`

File: appengine/standard/endpoints/backend/main.py [hunk 4]
```diff
@@ -22,128 +22,46 @@
...
     message = messages.StringField(1)
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/main.py]: how about for disambiguation we call this 'content' instead?

File: appengine/standard/endpoints/backend/main_test.py [hunk 5]
```diff
@@ -12,42 +12,43 @@
...
 # limitations under the License.
 
+import endpoints
 import main
 import mock
+import unittest
+from google.appengine.ext import testbed
 from protorpc import message_types
 
+class EchoTestCase(unittest.TestCase):
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/main_test.py]: please use pytest style tests. Refer to the test you overwrote here for an example.

File: appengine/standard/endpoints/backend/app.yaml (hunk 1 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/app.yaml]: There are, I don't have all the details around it, but once I do I'll be sure to test/update this pull.

Maybe @tbetbetbe can offer some more information.

Comment (jonparrott) [inline appengine/standard/endpoints/backend/app.yaml]: SG.

File: appengine/standard/endpoints/backend/app.yaml (hunk 2 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/app.yaml]: Done

File: appengine/standard/endpoints/backend/main.py (hunk 3 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/main.py]: Done

File: appengine/standard/endpoints/backend/main.py (hunk 4 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/main.py]: Done

File: appengine/standard/endpoints/backend/main_test.py (hunk 5 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/main_test.py]: Done

File: appengine/standard/endpoints/backend/echo-v1_swagger.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (jonparrott) [inline appengine/standard/endpoints/backend/echo-v1_swagger.json]: In the flexible endpoints sample, we use yaml for the definition. Is it possible to use yaml here as well or not?

**Comment by jonparrott** [comment]
I know this goes back on what I said initial, but since this will break a lot of the current docs pages can I recommend that you place this in `/appengine/standard/endpoints-v1.1/backend` instead of replacing the existing sample?

File: appengine/standard/endpoints/backend/echo-v1_swagger.json (hunk 6 above)
Comment (saicheems) [inline appengine/standard/endpoints/backend/echo-v1_swagger.json]: It is possible, I think. This swagger is autogenerated from the endpoints service by a new tool. I don't think it has an option to generate yaml, but I can convert it if that's preferred.

Comment (jonparrott) [inline appengine/standard/endpoints/backend/echo-v1_swagger.json]: If it's autogenerated don't sweat it.

**Comment by saicheems** [comment]
Done

**Comment by saicheems** [comment]
Redone against master.

**Pull request closed without merging**